extensions/pyRevitDevTools.extension/pyRevitDev.tab/Debug.panel/Unit Tests.pulldown/Test Family API.pushbutton/script.py

```diff
@@ -0,0 +1,156 @@
+"""
+Exercise the Revit family creation API:
+1. Creates a new Generic Annotation family and load it into the current document.
+2. Re-open the family, add some types, and re-load it.
+3. Check that the types were loaded, and the parameter values are correct.
+4. Purge the test family from the current document.
+
+Enable pyRevit log messages to see the test confirmation messages.
+Test errors are raised Python exceptions.
+"""
+import os
+import tempfile
+
+from pyrevit import revit, DB, script, HOST_APP
+from pyrevit.revit.db.create import FamilyLoaderOptionsHandler
+from pyrevit.revit.db.transaction import Transaction
+
+
+__context__ = "doc-project"
+
+TEMP_FAMILY_NAME = "pyRevitTestFamily"
+TEST_PARAMETER_NAME = "Test Parameter"
+TYPES_TO_ADD = {"A": 1.0, "B": 2.0, "C": 3.0, "D": 4.0}
+LOG = script.get_logger()
+
+
+def _purge_family():
+    """Remove the temporary family from the document."""
+    to_delete = None
+    collector = DB.FilteredElementCollector(revit.doc).OfClass(DB.Family)
+    for family_element in collector.GetElementIterator():
+        if family_element.Name == TEMP_FAMILY_NAME:
+            to_delete = family_element.Id
+            break
+    if to_delete is not None:
+        message = "Purge {0} family".format(TEMP_FAMILY_NAME)
+        with Transaction(message, doc=revit.doc):
+            revit.doc.Delete(to_delete)
+            LOG.info(message)
+
+
+def _family_type_count(family_id):
+    """
+    Return the number of types in a loaded family.
+    """
+    if isinstance(family_id, int):
+        family_id = DB.ElementId(family_id)
+    elif not isinstance(family_id, DB.ElementId):
+        raise TypeError()
+    family = revit.doc.GetElement(family_id)
+    return len(list(family.GetFamilySymbolIds()))
+
+
+def _add_types(family_doc):
+    """
+    Create new types in the test family, and assign values to the type
+    parameter.
+    """
+    if not isinstance(family_doc, DB.Document):
+        raise TypeError()
+    if not family_doc.IsFamilyDocument:
+        raise ValueError()
+    manager = family_doc.FamilyManager
+    parameter = manager.get_Parameter(TEST_PARAMETER_NAME)
+    with Transaction("Add Family Types", doc=family_doc):
+        for k, v in TYPES_TO_ADD.items():
+            manager.NewType(k)
+            manager.Set(parameter, v)
+
+
+def _check_type_parameter_values(family_id):
+    """
+    Check the family types and parameter values in the loaded test family.
+    """
+    if isinstance(family_id, int):
+        family_id = DB.ElementId(family_id)
+    elif not isinstance(family_id, DB.ElementId):
+        raise TypeError()
+    family_element = revit.doc.GetElement(family_id)
+    for type_id in family_element.GetFamilySymbolIds():
+        type_element = revit.doc.GetElement(type_id)
+        type_name = type_element.LookupParameter("Type Name").AsString()
```

Comment:
LookupParameter() won't work on non English Revit

Reply:
top of my head this should do: type_element.get_Parameter(DB.BuiltInParameter.ALL_MODEL_TYPE_NAME).AsString()

Reply:
top of my head this should do:` type_element.get_Parameter(DB.BuiltInParameter.ALL_MODEL_TYPE_NAME).AsString()`